commit b028b29dc9f4b67e55cc8e0e91d2c0e2457564ed
Author: Steve Dickson <steved@redhat.com>
Date:   Thu May 10 16:02:15 2007 -0400

    Added missing unlock_mtab() call in the add_mtab()
    routine.
    
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/mount/mount.c b/utils/mount/mount.c
index 3831415..d3156b2 100644
--- a/utils/mount/mount.c
+++ b/utils/mount/mount.c
@@ -211,6 +211,7 @@ int add_mtab(char *fsname, char *mount_point, char *fstype, int flags, char *opt
 	lock_mtab();
 
         if ((mtab = setmntent(MOUNTED, "a+")) == NULL) {
+		unlock_mtab();
 		fprintf(stderr, "Can't open " MOUNTED);
 		return 1;
 	}
